src/lib/crypto/kimchi_backend/gadgets/foreign_field.mli

```diff
@@ -221,6 +322,7 @@ val sub :
 val mul :
      (module Snark_intf.Run with type field = 'f)
   -> 'f External_checks.t (* external_checks *)
+  -> ?bound_check_result:bool
```

Comment:
I think so.  It's an option that's defaulted to off, but it can be very helpful for performance to use it when you know the result is already constrained.